store/sqlstore/channel_store.go

```diff
@@ -3428,29 +3443,59 @@ func (s SqlChannelStore) SearchGroupChannels(userId, term string) (model.Channel
 	return groupChannels, nil
 }
 
-// TODO: rewrite in squirrel (https://github.com/mattermost/mattermost-server/issues/19336)
-func (s SqlChannelStore) GetMembersByIds(channelId string, userIds []string) (model.ChannelMembers, error) {
-	var dbMembers channelMemberWithSchemeRolesList
+func (s SqlChannelStore) GetMembersByIds(channelID string, userIDs []string) (model.ChannelMembers, error) {
+	if len(userIDs) == 0 {
+		return nil, errors.New("no user IDs were provided")
+	}
 
-	keys, props := MapStringsToQueryParams(userIds, "User")
-	props["ChannelId"] = channelId
+	query := channelMembersForTeamWithSchemeSelectQueryX.Where(
+		sq.Eq{
+			"ChannelMembers.ChannelId": channelID,
+			"ChannelMembers.UserId":    userIDs,
+		},
+	)
+	if s.DriverName() == model.DatabaseDriverPostgres {
+		query = query.PlaceholderFormat(sq.Dollar)
+	}
+
+	sql, args, err := query.ToSql()
+	if err != nil {
+		return nil, errors.Wrap(err, "GetMembersByIds_ToSql")
+	}
+
+	var dbMembers channelMemberWithSchemeRolesList
```

Comment:
`dbMembers := channelMemberWithSchemeRolesList{}`

Reply:
This would slightly change the behavior, since it initializes the list instead of just leaving it `nil`. I'll change it if required.

Reply:
Yes, this is intentional. Squirrel throws an error in some cases if we pass nil. An empty slice is better to use.